### PR TITLE
Regions

### LINKED ISSUE #8: Intent to implement: State-specific holidays
In [some countries](https://www.fairwork.gov.au/employment-conditions/public-holidays/2023-public-holidays) each state decides their holidays more or less independently. Would you be interested in a PR that adds an optional state selector for those?

### PR DESCRIPTION
**Pull request by koterpillar**
Fixes #8.

### CODE REVIEW

File: js/holidays.js [hunk 1]
```diff
@@ -353,6 +355,21 @@ function getMonthDays( year ) {
...
 }
 
+/**
+ * Returns the holiday only if it's applicable in the current region
+ *
+ * @param {array} holidayRegions Regions the holiday is observed in
+ * @param {object} holiday Holiday definition
+ * @returns Array (zero or one) of holiday definitions applicable
+ */
+function inRegions ( holidayRegions, holiday ) {
```
Comment (hvianna) [inline js/holidays.js:365]: Would you mind adjusting the spacing in this block? Sorry if I adopt an unusual style, but the extra spacing really improves the code readability for me.
Could also go without the curly brackets in the if-else statements, but it's your call.
```
function inRegions( holidayRegions, holiday ) {
	if ( holidayRegions.includes( region ) ) {
		return [ holiday ];
	}
	else {
		return [];
	}
}
```

**Comment by hvianna** [comment]
This looks great! Thanks for the PR!

File: js/holidays.js (hunk 1 above)
Comment (koterpillar) [reply js/holidays.js:365]: Reformatted, sorry for slipping.

I'll leave the curly brackets in [just](https://www.imperialviolet.org/2014/02/22/applebug.html) in [case](https://www.visualmicro.com/page/User-Guide.aspx?doc=Curly-Braces-Ifs.html).

Comment (hvianna) [reply js/holidays.js:365]: Thanks! I know it's a terrible practice, but old habits die hard.. 😅

**Pull request merged**